src/SPC/builder/unix/library/imap.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SPC\builder\unix\library;
+
+use SPC\exception\FileSystemException;
+use SPC\exception\RuntimeException;
+use SPC\exception\WrongUsageException;
+
+trait imap
+{
+    /**
+     * @throws FileSystemException
+     * @throws RuntimeException
+     * @throws WrongUsageException
+     */
+    protected function build(): void
+    {
+        // ext-imap is NOT thread safe, do not build with ZTS builds
+        if ($this->builder->getOption('enable-zts')) {
+            return;
+        }
+        if ($this->builder->getLib('openssl') !== null) {
+            $this->replace_file_contents(
+                $this->source_dir . '/Makefile',
+                'SSLINCLUDE=/usr/include/openssl',
+                'SSLINCLUDE=' . BUILD_INCLUDE_PATH
+            );
+            $this->replace_file_contents(
+                $this->source_dir . '/Makefile',
+                'SSLLIB=/usr/lib',
+                'SSLLIB=' . BUILD_LIB_PATH
+            );
+            $this->replace_file_contents(
+                $this->source_dir . '/Makefile',
+                '"-DMAC_OSX_KLUDGE=1',
+                ''
+            );
+            $this->replace_file_contents(
+                $this->source_dir . '/src/osdep/unix/Makefile',
+                '-lcrypto -lz',
+                '-lcrypto'
+            );
+            $this->replace_file_contents(
+                $this->source_dir . '/src/osdep/unix/Makefile',
+                '-lcrypto',
+                '-lcrypto -lz'
```

Comment:
We have `FileSystem::replaceFileStr('/path/to/file', 'src str', 'dst str');`. See https://github.com/crazywhalecc/static-php-cli/blob/main/src/SPC/store/FileSystem.php

Reply:
Thank you, will refactor to use that